src/GraphQLParser/Token.cs

```diff
@@ -1,24 +1,27 @@
+using System;
+
 namespace GraphQLParser
 {
     public readonly struct Token
     {
-        public Token(TokenKind kind, string? value, int start, int end)
+        public Token(TokenKind kind, ReadOnlyMemory<char> value, int start, int end)
         {
             Kind = kind;
             Value = value;
             Start = start;
             End = end;
-        }
 
-        public int End { get; }
+        }
 
         public TokenKind Kind { get; }
 
         public int Start { get; }
 
-        public string? Value { get; }
+        public int End { get; }
```

Comment:
I had an idea to replace `int` with `short` to reduce the size of the structure, but then long queries will stop working. It would be possible to solve this problem somehow using a generic parameter, but I did not complicate the code.

Reply:
I would leave it as an `int`.